### ISSUE TITLE
Crash on save state

### ISSUE THREAD

**Issue by claude13112001** [hrydgard/ppsspp#7862]
Hi all,

(Maybe you remember me,maybe not,if not it's probably because I haven't any problem with this great emulator since now. I am trying to play all GTA 3 era games 100% which LCS and VCS are parts of too so i play them (just LCS for now) on your emulator.) 

ISSUE: The problem is that when I save state,the emulator crashes and I have to reload it and save state many times,but when it saves once,then for that play session (until quiting the game) it doesn't crash,but the next time,it's the same. It never happened when I played on versions under 1.0 and before my bigger "pause" with the emulator. But now,I also tried older versions and it happens in them too now. I don't know why it's happening,maybe it's because of my settings? 

(I'll post them later (along with anything needed, that you'll ask me for to solve the problem) and some screens if you'll need them.)

Thanks for any replies :)

**Comment by unknownbrackets** [comment]
What device is this?

The main reason the emulator might crash when saving state is if your device runs out of memory.  This would mainly be likely on Symbian or iOS devices.

-[Unknown]

**Comment by claude13112001** [comment]
It's PC,Windows 7 Professional.

**Comment by unknownbrackets** [comment]
Well, that's weird.

Is it able to load the savestate it created during the crash?  Or is it an older savestate?

Just to be sure, I assume on a PC you have free memory right?  And free disk space?

Does this happen with OpenGL and Direct3D 9 backends?  It's possible there's some interaction with the video driver, but I'm not sure this is likely.

-[Unknown]

**Comment by claude13112001** [comment]
But it crashes just the first time for a playing session,when I save it first time,for the whole playing session I'm good to go. 

And I have about 9 GB out of 250 GB free but I plan to free up some space :D

It happens on both backends.

**Comment by claude13112001** [comment]
Is somebody "testing" if he/she have the same problem?

**Comment by claude13112001** [comment]
I tried it in GTA CHW and it didn't crash,so maybe it was just an accident or GTA LCS is the only (or one of few) that it happens with.

**Comment by claude13112001** [comment]
Okay,I am now playing GTA VCS and it does the same thing as in LCS.

**Comment by claude13112001** [comment]
You don't have any solutions or reasons why it could happen do you? I would appreciate some answer already.

**Comment by unknownbrackets** [comment]
I personally just do this in my free time for fun.  People bumping things without much useful info annoys me which ain't fun, so you'll understand if it makes me ignore uninteresting issues.

Memory isn't the same thing as disk space.  If you right click on Computer and go to Properties, you should see "Installed memory (RAM)".  It's unlikely you're running out of that on a desktop, but if that number is like 2.0 GB, it could be the reason it's crashing perhaps (or if you have a lot of programs open and it's higher, maybe?)  It's a situation we could handle better, although we typically only require ~128 MB to create a save state at most.

If it happens with both backends, this doesn't really disprove the video driver idea.  Have you made sure you've updated your drivers?  What card do you have?

I don't really understand what you're saying though.  "It crashes just the first time in a playing session."  To me a playing session is when the program is running.  If it crashes, the playing session is over.  Do you mean it crashes the first time you save state after restarting your computer or something?  Or just if you leave it alone for a while?  Those sound like potential problems with your computer or OS install, if so.

It sounds like this is only with GTA games, but you're not really being clear.  I don't have (or like) GTA, so I personally have not tested this.  However, the game is more or less popular, so I'd be surprised if other people had the same experience but no one else was reporting it.

-[Unknown]

**Comment by KmatToS** [comment]
Slightly different issue, didn't seem thread-worthy when this was open and recent.

I've been running Yu-Gi-Oh Tag Force GX on the 64-bit client, which runs very well for extended periods of time. I use save states frequently, but I've run into the issue twice were a save state I created moments ago will fail to load. The console isn't crashing or anything, I just open the save state screen when nothing is going on in the game, save state>load state>"failed to load save state". At that point it re-launches the game and I'm sent back to the last hard save. I have two save states from different play attempts on the system now. I don't know they can be used for log dumps or anything. I've only been using ppsspp for a few days, but nothing I've read online seems to address this.

**Comment by unknownbrackets** [comment]
@KmatToS not sure what PPSSPP version you're using.  Not that long ago (past year maybe?), we changed it to save to a temp file, and then swap in the completed save.  This prevents corrupting a savestate if you run out of disk space or hit some bug during save.

If you're losing savestates and there's no error message, it sounds like you're not using the latest version.

The most common situation iirc is disk full, but that's on mobile.  Usually disk full won't happen on desktop unless you're saving to a usb stick or something.  Another case is antivirus - an antivirus may false-positive match the savestate file and delete/modify it.  It depends on the antivirus how well this is handled by PPSSPP, unfortunately.  That should be rare, though.

-[Unknown]

**Comment by KmatToS** [comment]
None of those sound like anything that could happen to me. I'm using the 1.0.1 version, says it's created February 26. I didn't see any updates since then.

**Comment by KmatToS** [comment]
Here's a pic of the console from when I go to load the state that fails.
![ppsspp savestate fail console](https://cloud.githubusercontent.com/assets/13190374/9242210/03cc8130-414c-11e5-8fbb-9741fa05a215.png)

**Comment by unknownbrackets** [comment]
I wonder if we're looking at a Snappy compression bug.

-[Unknown]

**Comment by KmatToS** [comment]
Maybe it happens because I'm overwriting the same save state too quickly?
I'll try fiddling with it and see if I can make it bork again.

**Comment by claude13112001** [comment]
That one with the playing session I mean like,if I success save state-ing the first time in playing session,it won't (at least didn't) crash anymore until I restart the emulator (maybe even if I restart the game,haven't tried that).

**Comment by DiegoSnip** [comment]
Same problem here with Metal Gear Portable Ops. I figured out that it doesn't crash if I use the File -> Save state (F2) function (you can access to this through ESC key while in the game). If I use the key bound to Save State in the key mapping config (F8 in my case), it crashes.

**Comment by daniel229** [comment]
I just crash once by pressing F2
![03](https://cloud.githubusercontent.com/assets/3481559/11199848/5378f144-8d0c-11e5-91db-67c6dfc489ca.png)

**Comment by daniel229** [comment]
Don't know why the save broken suddenly.
![02](https://cloud.githubusercontent.com/assets/3481559/11317685/014711e0-9071-11e5-8afa-5a059a0f5333.png)

**Comment by unknownbrackets** [comment]
I've never been able to experience a crash save stating.  Can anyone reproduce this with any consistency?  If so, anyone have the ability to get a stack trace?

-[Unknown]

**Comment by unknownbrackets** [comment]
> if I success save state-ing the first time in playing session,it won't (at least didn't) crash anymore until I restart the emulator (maybe even if I restart the game,haven't tried that).

#8471 very possibly fits this description.

But it doesn't describe why "IntrHandler" might cause crashes (seen in the other two issues.)  IntrHandler is saved right after sceMp4, which is not used by many games and doesn't look to be in use in the two screenshots above.

It might be possible to investigate that further with a broken savestate file uploaded.

If we're not seeing savestates break anymore after #8471, we might close this and reopen when we see other issues.

-[Unknown]

**Comment by unknownbrackets** [comment]
No new feedback in a while.  Closing for now, but if more issues refurface we can reopen or create a new issue.

-[Unknown]
